### PR TITLE
Latest releases of Camp, RAJA, Umpire, CHAI and CARE

### PR DESCRIPTION
**Pull request by adrienbernede**
Add latest releases to camp, raja and umpire
Add Adrien Bernede as package maintainer

### CODE REVIEW

**Comment by adayton1** [comment]
Would you like me to add new release info for chai and care to this merge request?

File: packages/camp/package.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import os
+    depends_on("blt@0.6.2:", type="build", when="@2024.02.1:")
```
Comment (white238) [inline packages/camp/package.py:7]: I'm gonna guess this isnt right =)

Comment (adrienbernede) [reply packages/camp/package.py:7]: I need automated testing in that repo, my copy-paste skills are dubious...

**Comment by adrienbernede** [comment]
> Would you like me to add new release info for chai and care to this merge request?

I created this PR after pushing the same changes to Spack. Could you instead push to Spack the relevant changes for CHAI and CARE ?
It is annoying, but since we have local diffs with Spack packages, it’s easier to push generic changes to Spack and then apply them in Radiuss-spack-configs, although things are supposed to happen the other way.

**Review by white238 — APPROVED** [review]

File: packages/camp/package.py [hunk 2]
```diff
@@ -136,11 +142,14 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("cub", when="+cuda")
 
     depends_on("blt", type="build")
+    depends_on("blt@0.6.2:", type="build", when="@2024.02.1:")
     depends_on("blt@0.6.1:", type="build", when="@2024.02.0:")
     depends_on("blt@0.5.0:0.5.3", type="build", when="@2022.03.0:2023.06.0")
 
     patch("libstdc++-13-missing-header.patch", when="@:2022.10")
 
+    patch("camp-rocm6.patch", when="@0.2.3 +rocm ^hip@6:")
```
Comment (white238) [inline packages/camp/package.py:151]: This patch does not exist in this repo.

Comment (white238) [reply packages/camp/package.py:151]: nm.. i am ansy and pushed it up.

Comment (adrienbernede) [reply packages/camp/package.py:151]: Thank you Chris.

**Comment by adrienbernede** [comment]
This is causing CHAI@develop to fail. I’d like to clarify this before merging.

**Comment by adayton1** [comment]
> > Would you like me to add new release info for chai and care to this merge request?
> 
> I created this PR after pushing the same changes to Spack. Could you instead push to Spack the relevant changes for CHAI and CARE ? It is annoying, but since we have local diffs with Spack packages, it’s easier to push generic changes to Spack and then apply them in Radiuss-spack-configs, although things are supposed to happen the other way.

Here's a merge request in Spack for updates to the chai package: https://github.com/spack/spack/pull/44911

**Comment by balos1** [comment]
Here is a PR in Spack for updates to care: https://github.com/spack/spack/pull/44936.

File: packages/camp/package.py [hunk 3]
```diff
@@ -103,11 +103,17 @@ class Camp(CMakePackage, CudaPackage, ROCmPackage):
...
 
-    maintainers("trws")
+    maintainers("trws", "adrienbernede")
 
     license("BSD-3-Clause")
 
     version("main", branch="main", submodules=False)
+    version(
+        "2024.02.1",
+        tag="v2024.02.",
```
Comment (MrBurmark) [inline packages/camp/package.py:113]:
```suggestion
        tag="v2024.02.1",
```

**Comment by adrienbernede** [comment]
@balos1 added changes from CARE, now testing...
@white238 waiting for patch to fix CHAI testing before merging...

File: packages/raja/package.py [hunk 4]
```diff
@@ -195,6 +208,7 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("camp+openmp", when="+openmp")
     depends_on("camp@main", when="@develop")
     depends_on("camp@main", when="@main")
+    depends_on("camp@2024.02.1:", type="build", when="@2024.02.1:")
```
Comment (rhornung67) [inline packages/raja/package.py:211]: @adrienbernede can we get the changes in the other PR pulled into this one: https://github.com/LLNL/radiuss-spack-configs/pull/106

The references to branch dependencies break our CI due to the way Spack does mirroring. @white238 can explain the details better than me. The other PR fixes the issue.

We are preparing new releases for RAJA, Umpire, and CHAI and will have new version info for those.

Comment (white238) [reply packages/raja/package.py:211]: I suspect the usage of branches in combination of a source mirror was causing Spack to lock down the git branch mirror to an old commit that wasn't updating. I have noticed this behavior before when you add a package to a spack mirror that uses a branch, Spack doesn't track the commit moved forward, just that the branch exists in the repo and will not update it. So @rhornung67  was getting and old camp.

Comment (adrienbernede) [reply packages/raja/package.py:211]: But... @white238 I don’t remember using a source mirror... does it apply to binary caches too ?

Comment (white238) [reply packages/raja/package.py:211]: My guess is it applies to both. It has to do with the fact that Spack treats branch names as versions. So the branch name gets locked down at a first come first serve basis.

**Comment by adrienbernede** [comment]
@adayton1 @davidbeckingsale I’m hitting a new error with fmt in CHAI CI.
Note that with the update of Spack, the CI is now using fmt@11.0.0.
I applied the same patch we had for fmt@10.x.x to fix the CI in Umpire.
However, I tested with fmt@10.2.1 and see the same error in CHAI CI, so that one is not specific to the fmt update.

**Review by rhornung67 — APPROVED** [review]
@adrienbernede Is this ready to merge? We need some of the spack package changes in here for the RAJA release, which is imminent. Thank you.

File: packages/umpire/package.py [hunk 5]
```diff
@@ -455,6 +462,9 @@ def initconfig_package_entries(self):
         else:
             entries.append(cmake_cache_option("ENABLE_IPC_SHARED_MEMORY", "+ipc_shmem" in spec))
 
+        if "~fmt_header_only" in spec:
+            entries.append(cmake_cache_string("UMPIRE_FMT_HEADER", "fmt::fmt"))
```
Comment (adayton1) [inline packages/umpire/package.py:466]: The CMake option is UMPIRE_FMT_TARGET, not UMPIRE_FMT_HEADER.

File: packages/chai/package.py [hunk 6]
```diff
@@ -136,6 +150,8 @@ class Chai(CachedCMakePackage, CudaPackage, ROCmPackage):
         depends_on("umpire+cuda")
         for sm_ in CudaPackage.cuda_arch_values:
             depends_on("umpire+cuda cuda_arch={0}".format(sm_), when="cuda_arch={0}".format(sm_))
+        with when("@2024.02.0:"):
+            depends_on("umpire~fmt_header_only")
```
Comment (adayton1) [inline packages/chai/package.py:154]: I was under the impression that the header only version works better with CUDA. I might be remembering incorrectly, though.

Comment (davidbeckingsale) [reply packages/chai/package.py:154]: That's the default which was not working, so I was going to try with the library version.

File: packages/raja/package.py [hunk 7]
```diff
@@ -192,11 +205,12 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on("blt@0.3.6:0.4.1", type="build", when="@:0.12.0")
     conflicts("^blt@:0.3.6", when="+rocm")
 
+    depends_on("camp",type="build")
```
Comment (adayton1) [inline packages/raja/package.py:208]: Why is the type set to "build"?

Comment (rhornung67) [reply packages/raja/package.py:208]: I got that from @white238. I think it's just a generic way of saying that camp is part of RAJA's build stack.

Comment (adayton1) [reply packages/raja/package.py:208]: I thought that was a way of marking dependencies that aren't libraries (e.g. cmake, blt).

Comment (davidbeckingsale) [reply packages/raja/package.py:208]: Could be a relic from when camp was header only?

Comment (white238) [reply packages/raja/package.py:208]: The types are `build`, `link`, `run`, and `test`. `build` by itself is not correct here.

Comment (rhornung67) [reply packages/raja/package.py:208]: So replace with ``depends_on("camp")`` ??

Comment (adayton1) [reply packages/raja/package.py]: Fixed.

**Review by adayton1 — APPROVED** [review]
LGTM

**Comment by rhornung67** [comment]
Thank you @adrienbernede

File: packages/raja/package.py (hunk 7 above)
Comment (rhornung67) [reply packages/raja/package.py]: Thank you @adayton1

**Pull request merged**